### PR TITLE
[DEV-1358]: add strapi configuration for hero slider

### PR DESCRIPTION
**Pull request by MikeAtUqido**
#### List of Changes

 * Add Hero schema to Strapi CMS

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/strapi-cms/src/components/common/cta.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "collectionName": "components_common_ctas",
+  "info": {
+    "displayName": "HeroCardCta",
```
Comment (marcobottaro) [inline apps/strapi-cms/src/components/common/cta.json:4]: I would use a more generic CallToAction component (maybe using the Link component)

File: apps/strapi-cms/src/components/common/hero.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "options": {},
+  "attributes": {
+    "siteTitle": {
+      "type": "string",
+      "required": true
+    },
+    "boldTitle": {
+      "type": "string",
+      "required": true
+    },
```
Comment (marcobottaro) [inline apps/strapi-cms/src/components/common/hero.json:16]: The hero doesn't have a `siteTitle` and a `boldTitle`. I would move both to the Card entity and join them in a single wysiwyg field

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/lib/strapi/__tests__/homepage.test.ts [hunk 3]
```diff
@@ -44,6 +44,15 @@ const makeStrapiResponseJson = () => ({
...
       },
+      hero: {
+        id: 2,
+        siteTitle: 'Titolo',
+        boldTitle: 'ciao',
+        cards: [
+          { id: 2, title: 'Prova', color: null },
+          { id: 6, title: 'Prova 2', color: null },
+        ],
+      },
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/__tests__/homepage.test.ts:55]: You should use phrases and words similar to those in the previous lines

File: apps/nextjs-website/src/lib/strapi/homepage.ts [hunk 4]
```diff
@@ -22,6 +41,11 @@ export const StrapiHomepageCodec = t.strict({
         title: t.string,
         links: t.array(LinkHomepageCodec),
       }),
+      hero: t.type({
+        siteTitle: t.string,
+        boldTitle: t.string,
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/homepage.ts:46]: Remove

File: apps/nextjs-website/src/lib/strapi/homepage.ts [hunk 5]
```diff
@@ -34,6 +58,9 @@ const makeStrapiHomepagePopulate = () =>
       comingsoonDocumentation: {
         populate: ['links'],
       },
+      hero: {
+        populate: ['boldTitle', 'siteTitle', 'cards'],
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/homepage.ts:62]:
```suggestion
        populate: ['cards'],
```

**Review by kin0992 — COMMENTED** [review]
you should add the changeset

File: package.json [hunk 6]
```diff
@@ -17,7 +17,7 @@
   },
   "devDependencies": {
     "@changesets/cli": "^2.27.1",
-    "husky": "^8.0.3",
+    "husky": "^9.0.11",
```
Comment (kin0992) [inline package.json:20]: why you need to bump husky?

File: package.json [hunk 7]
```diff
@@ -5,7 +5,7 @@
     "packages/*"
   ],
   "scripts": {
-    "prepare": "husky install",
+    "prepare": "husky",
```
Comment (kin0992) [inline package.json:8]: why this PR should handle anything related the `husky` dependency?

File: package.json (hunk 6 above)
Comment (MikeAtUqido) [reply package.json:20]: The husky version was causing a bug that prevented the install of the `@strapi/blocks-react-renderer` lib that's used by the frontend to render the blocks elements fetched from strapi like images, the error was:
Screenshot 2024-02-20 alle 09 48 35

Comment (kin0992) [reply package.json:20]: I'm not sure the bump is necessary: I just tried to run the command you executed in the screenshot starting from main, and it succeeded. Could be some misconfiguration you have on your machine?

File: apps/nextjs-website/src/components/atoms/CtaSlide/CtaSlide.tsx [hunk 8]
```diff
@@ -41,37 +49,45 @@ const CtaSlide = ({
...
-            xs: 'clamp(36px, 8vw, 48px)',
-            md: 'clamp(48px, 10vw, 58px)',
-          }}
-          variant={'h1'}
-          color={textColor}
-        >
-          {title}
-        </Typography>
+        {fetchFromStrapi ? (
+          <BlockRendererClient content={subhead as BlocksContent} />
```
Comment (datalek) [inline apps/nextjs-website/src/components/atoms/CtaSlide/CtaSlide.tsx:53]: Do not cast, fix the type instead casting

File: apps/nextjs-website/src/config.ts [hunk 9]
```diff
@@ -23,6 +28,16 @@ export const amplifyConfig = {
...
+// a BuildEnv instance ready to be used
+export const buildEnv = pipe(
+  makeBuildConfig(process.env),
+  E.map(makeBuildEnv),
+  E.getOrElseW((errors) => {
+    // eslint-disable-next-line functional/no-throw-statements
+    throw errors;
+  })
+);
+
```
Comment (kin0992) [inline apps/nextjs-website/src/config.ts:40]: This file is going to be deprecated (as the comment on top of the file says), so you should not add stuff here.

Moreover, the `buildEnv` is an environment used during build time, so I think that if you use it on client side, this doesn't work as expected
```suggestion
```

File: apps/nextjs-website/src/components/atoms/CtaSlide/CtaSlide.tsx [hunk 10]
```diff
@@ -4,28 +4,36 @@ import { ButtonNaked } from '@pagopa/mui-italia';
...
+  readonly color?: string | null;
+  readonly variant?: 'text' | 'contained' | 'outlined' | null;
   readonly cta?: {
-    readonly label: string;
+    readonly text: string;
     readonly href: string;
-    readonly variant?: 'text' | 'contained' | 'outlined';
-  };
+    readonly target?: '_self' | '_blank' | '_parent' | '_top' | null;
+  } | null;
```
Comment (datalek) [inline apps/nextjs-website/src/components/atoms/CtaSlide/CtaSlide.tsx:19]: Are the `null` really required?

**Review by datalek — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/atoms/CtaSlide/CtaSlide.tsx [hunk 11]
```diff
@@ -41,37 +49,45 @@ const CtaSlide = ({
...
+            fontSize={{
+              xs: 'clamp(36px, 8vw, 48px)',
+              md: 'clamp(48px, 10vw, 58px)',
+            }}
+            variant={'h1'}
+            color={textColor}
+          >
+            {subhead as string}
+          </Typography>
+        )}
```
Comment (kin0992) [inline apps/nextjs-website/src/components/atoms/CtaSlide/CtaSlide.tsx:65]: Why this logic is inside the component?
The component should behave the same if the data are coming from Strapi or from static contents; the component should receive information, typed as props, and render them

**Review by kin0992 — CHANGES_REQUESTED** [review]

**Comment by marcobottaro** [comment]
Replaced by #671 and #672

**Pull request closed without merging**